http_parser.c

```diff
@@ -2326,7 +2326,6 @@ http_parse_host(const char * buf, struct http_parser_url *u, int found_at) {
     case s_http_host_v6:
     case s_http_host_v6_zone_start:
     case s_http_host_v6_zone:
-    case s_http_host_port_start:
```

Comment:
This changes the behavior of `http_parser_parse_url(is_connect=1)` in that it'll now allow `CONNECT hostname:` as valid input, doesn't it?